make condition events use RegisterUnitEvent when possible

mrbuds opened the pull request:
this should fix performance issues with the cooldown trigger with "Insufficient Resources" and "Spell Usable" conditions
add support for filtering in custom checks too
does not support multi units (group, nameplate, ...)

**This will break existing custom checks conditions using ":" as separator**

InfusOnWoW commented on WeakAuras/Conditions.lua:
```diff
@@ -781,14 +797,29 @@ function Private.RegisterForGlobalConditions(uid)
 function Private.UnregisterForGlobalConditions(uid)
   for event, condFuncs in pairs(dynamicConditions) do
     condFuncs[uid] = nil;
+    if next(dynamicConditions) == nil then
```
I this should be next(dynamicConditions)

InfusOnWoW commented on WeakAuras/Conditions.lua:
```diff
@@ -781,14 +797,29 @@ function Private.RegisterForGlobalConditions(uid)
+    if next(dynamicConditions) == nil then
+      if not event:match("([^:]+):([^:]+)") then
+        dynamicConditionsFrame:UnregisterEvent(event)
+      end
```
Was there a reason you didn't add unregistering for the unit events?

mrbuds commented:
@InfusOnWoW https://github.com/WeakAuras/WeakAuras2/pull/3470/commits/36529523d81438378be2e188a0497e8b02b3517f should fix your comments

InfusOnWoW commented:
We should actually also unregister in UnloadAllConditions

mrbuds commented:
> We should actually also unregister in UnloadAllConditions

fixed, and also fixed a bug in UnregisterForGlobalConditions not cleaning dynamicConditions that prevented events to be registered again
